test/integration/node-specific/client_close.test.ts

```diff
@@ -128,25 +329,138 @@ describe.skip('MongoClient.close() Integration', () => {
+                  data: {
+                    blockConnection: true,
+                    blockTimeMS: waitQueueTimeoutMS * 3,
+                    failCommands: ['insert']
+                  }
+                };
+                await utilClient.db('admin').command(failPoint);
+
+                const timeoutStartedSpy = sinon.spy(timers, 'setTimeout');
+
+                const client = new MongoClient(uri, {
+                  minPoolSize: 1,
+                  maxPoolSize: 1,
+                  waitQueueTimeoutMS
+                });
+                const insertPromise = client
+                  .db('db')
+                  .collection('collection')
+                  .insertOne({ x: 1 })
+                  .catch(e => e);
+                client
+                  .db('db')
+                  .collection('collection')
+                  .insertOne({ x: 1 })
+                  .catch(e => e);
+
+                // don't allow entire checkout timer to elapse to ensure close is called mid-timeout
+                await sleep(waitQueueTimeoutMS / 2);
+                const checkoutTimeoutStarted =
+                  timeoutStartedSpy
+                    .getCalls()
+                    .filter(r => r.args.includes(waitQueueTimeoutMS))
+                    .flat().length > 0;
+                expect(checkoutTimeoutStarted).to.be.true;
+
+                await client.close();
+                expect(getTimerCount()).to.equal(0);
+                // un-configure fail{oint
+                await utilClient.db().admin().command({
+                  configureFailPoint: 'failCommand',
+                  mode: 'off'
+                });
+
+                await utilClient.close();
+
+                const err = await insertPromise;
+                expect(err).to.be.instanceOf(Error);
+                expect(err.message).to.contain(
+                  'Timed out while checking out a connection from connection pool'
+                );
+              };
+              await runScriptAndGetProcessInfo('timer-check-out', config, run);
+            });
           });
         });
 
         describe('Connection', () => {
           describe('Node.js resource: Socket', () => {
-            describe('after a connection is checked out', () => {
-              it.skip('no sockets remain after client.close()', async () => {});
-            });
-
             describe('after a minPoolSize has been set on the ConnectionPool', () => {
-              it.skip('no sockets remain after client.close()', async () => {});
+              it('no sockets remain after client.close()', async function () {
+                const run = async function ({ MongoClient, uri, expect }) {
+                  const connectionMonitoringReport = () =>
+                    process.report.getReport().libuv.filter(r => r.type === 'tcp');
+
+                  // assert no sockets to start with
+                  expect(connectionMonitoringReport()).to.have.length(0);
```

Comment:
```suggestion
                  expect(connectionMonitoringReport()).to.have.lengthOf(0);
```

I could be wrong but I think you have to use lengthOf